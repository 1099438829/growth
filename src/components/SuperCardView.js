import React, { PropTypes } from 'react';
import {
  View, Text, StyleSheet, Platform,
  TouchableHighlight,
} from 'react-native';
import { Icon } from 'react-native-elements';
import Accordion from 'react-native-collapsible/Accordion';
import * as Animatable from 'react-native-animatable';

import AppColors from '../theme/colors';

const styles = StyleSheet.create({
  cardContainer: {
    backgroundColor: '#fff',
    borderTopLeftRadius: 6,
    borderTopRightRadius: 6,
    margin: 10,
    height: 100,
    ...Platform.select({
      ios: {
        shadowColor: 'rgba(0,0,0, .2)',
        shadowOffset: { height: 1, width: 0 },
        shadowOpacity: 1,
        shadowRadius: 1,
      },
      android: {
        elevation: 1,
      },
    }),
  },
});

const SECTIONS = [
  {
    title: 'First',
    content: 'Lorem ipsum...',
  },
  {
    title: 'Second',
    content: 'Lorem ipsum...',
  },
];

class SuperCardView extends React.PureComponent {

  static renderContent(section, i, isActive) {
    return (
      <Animatable.View
        duration={300}
        transition="backgroundColor"
        style={{ backgroundColor: (isActive ? 'rgba(255,255,255,1)' : 'rgba(245,252,255,1)') }}
      >
        <Animatable.Text
          duration={300}
          easing="ease-out"
          animation={isActive ? 'zoomIn' : false}
        >
          {section.content}
        </Animatable.Text>
      </Animatable.View>
    );
  }

  static renderHeader(section, index, isActive) {
    return (
      <Animatable.View
        duration={300}
        transition="backgroundColor"
      >
        <View style={styles.cardContainer}>
          <TouchableHighlight
            underlayColor="#fff"
            onPress={() => {}}
          >
            <View>
              <View style={{ flex: 1, flexDirection: 'row' }}>
                <View style={{ flex: 1, justifyContent: 'space-between', flexDirection: 'column', margin: 15 }}>
                  <View style={{ height: 40 }}>
                    <Text style={{ fontSize: 20, marginTop: 5, marginBottom: 15 }}>
                      {this.props.title}
                    </Text>
                  </View>
                  <View style={{ height: 40 }}>
                    <Text style={{ marginTop: 5, marginBottom: 15 }}>
                      {this.props.subTitle}
                    </Text>
                  </View>
                </View>
              </View>
              <View style={{ height: 100, alignItems: 'flex-end', left: -10 }}>
                <Icon size={100} name={this.props.iconName} type={'ionicon'} color={AppColors.brand.primary} />
              </View>
            </View>
          </TouchableHighlight>
        </View>
      </Animatable.View>
    );
  }

  constructor(props) {
    super(props);
    this.renderContent = SuperCardView.renderContent.bind(this);
    this.renderHeader = SuperCardView.renderHeader.bind(this);
  }

  componentName = 'SuperCardView';


  render() {
    return (
      <Accordion
        sections={SECTIONS}
        renderHeader={this.renderHeader}
        renderContent={this.renderContent}
      />
    );
  }
}


SuperCardView.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  iconName: PropTypes.string.isRequired,
};

export default SuperCardView;
