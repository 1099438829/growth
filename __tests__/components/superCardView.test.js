import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import SuperCardView from '../../src/components/SuperCardView';

it('renders correctly', () => {
  // eslint-disable-next-line no-unused-vars
  const tree = renderer.create(
    <SuperCardView title={'从零开始'} subTitle={'从这里准备开始你的技术图谱修炼之路'} iconName={'md-home'} />,
  );

  const treeJson = tree.toJSON();
  expect(treeJson).toMatchSnapshot();
});
