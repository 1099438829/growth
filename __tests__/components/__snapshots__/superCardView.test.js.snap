// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(238, 238, 238, 1.0)",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
              "height": 100,
              "margin": 10,
              "shadowColor": "rgba(0,0,0, .2)",
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "margin": 15,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 40,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 20,
                        "marginBottom": 15,
                        "marginTop": 5,
                      }
                    }
                  >
                    从零开始
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "height": 40,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "marginBottom": 15,
                        "marginTop": 5,
                      }
                    }
                  >
                    从这里准备开始你的技术图谱修炼之路
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "height": 100,
                  "left": -10,
                }
              }
            >
              <View
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#03a9f4",
                        "fontSize": 100,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        undefined,
                      ],
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={Object {}}
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(238, 238, 238, 1.0)",
              "opacity": 1,
            }
          }
        >
          <View
            style={Object {}}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <View
                onLongPress={undefined}
                onPress={undefined}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "#ededed",
                      "borderBottomWidth": 1,
                      "paddingBottom": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginLeft": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      style={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {},
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            Array [
                              Object {
                                "color": "#43484d",
                                "fontSize": 17.5,
                              },
                              Object {
                                "marginLeft": 10,
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        hello, world
                      </Text>
                    </View>
                    <View
                      style={undefined}
                    >
                      <View />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "flex": 0.15,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      onPress={undefined}
                      style={
                        Array [
                          false,
                          false,
                          false,
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                      underlayColor="white"
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#bdc6cf",
                              "fontSize": 28,
                            },
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                              },
                              undefined,
                            ],
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLongPress={undefined}
                onPress={undefined}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "#ededed",
                      "borderBottomWidth": 1,
                      "paddingBottom": 10,
                      "paddingRight": 10,
                      "paddingTop": 10,
                    },
                    undefined,
                  ]
                }
                underlayColor="white"
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "marginLeft": 10,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      style={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {},
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            Array [
                              Object {
                                "color": "#43484d",
                                "fontSize": 17.5,
                              },
                              Object {
                                "marginLeft": 10,
                              },
                              undefined,
                              undefined,
                            ],
                          ]
                        }
                      >
                        开发环境
                      </Text>
                    </View>
                    <View
                      style={undefined}
                    >
                      <View />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "flex": 0.15,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      onPress={undefined}
                      style={
                        Array [
                          false,
                          false,
                          false,
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          },
                          undefined,
                        ]
                      }
                      underlayColor="white"
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#bdc6cf",
                              "fontSize": 28,
                            },
                            Array [
                              Object {
                                "backgroundColor": "transparent",
                              },
                              undefined,
                            ],
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
